src/Microsoft.Azure.SignalR/EndpointProvider/ServiceEndpointManager.cs

```diff
@@ -58,6 +55,7 @@ private void OnChange(ServiceOptions options)
 
         private Task ReloadServiceEndpointsAsync(IEnumerable<ServiceEndpoint> serviceEndpoints)
         {
+            _synchronizer.OnChange(serviceEndpoints);
```

Comment:
I don't think you need to inject `synchronizer` into `ServiceEndpointManager`. You just need to inject `IOptionsMonitor<ServiceOptions>` into `AccessKeySynchronizer`, and register your `OnChange` listener.

Reply:
Asp.net and .net Core have different `ServiceOptions` respectively, it will require us to create different `synchronizer` for each framework if we did so. Thus I don't think it's a good idea.